cylc/rose/utilities.py

```diff
@@ -57,19 +57,16 @@ def get_rose_vars_from_config_node(config, config_node, environ):
 
     """
     templating = None
-    if (
-        'jinja2:suite.rc' in config_node.value and
-        'empy:suite.rc' in config_node.value
-    ):
+    sections = {'jinja2:suite.rc', 'empy:suite.rc', 'template variables'}
+    defined_sections = sections.intersection(set(config_node.value))
+    if len(defined_sections) > 1:
         raise MultipleTemplatingEnginesError(
-            "You should not define both jinja2 and empy in the same "
-            "configuration file."
+            "You should not define more than one templating section. "
+            f"You defined:\n\t{'; '.join(defined_sections)}"
         )
-    elif 'jinja2:suite.rc' in config_node.value:
-        templating = 'jinja2'
-    elif 'empy:suite.rc' in config_node.value:
-        templating = 'empy'
-    if templating:
+    elif len(defined_sections) == 1:
+        templating, = list(defined_sections)
+    if templating is not None:
         config['templating_detected'] = templating
```

Comment:
I think this messes with the reported value of `templating_detected`?

Possible fix:

```suggestion
    elif len(defined_sections) == 1:
        templating, = defined_sections
        if templating != 'template variables':
            config['templating_detected'] = templating.replace(':suite.rc', '')
```

Reply:
It does, I just changed the value of `templating_detected` all the way through and into cylc-flow. I think it makes the code easier to follow.